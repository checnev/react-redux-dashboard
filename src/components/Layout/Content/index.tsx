import React from 'react';
import {
  Container,
  Grid,
} from '@material-ui/core';

interface ContentProps {
  children: React.ReactNode;
  sidebar?: React.ReactNode;
}

const Content: React.FC<ContentProps> = ({ children, sidebar: SideBar}: ContentProps) => {
  const contentSize = SideBar? 9 : 12;

  return (
    <Container maxWidth="lg">
      <Grid container>
        { SideBar && <Grid item xs={12} sm={3}>{SideBar}</Grid> }
        <Grid item xs={12} sm={contentSize}>{children}</Grid>
      </Grid>
    </Container>
  );
};

export default Content;
