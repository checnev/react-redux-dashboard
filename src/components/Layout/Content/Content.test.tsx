import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Content from 'components/Layout/Content';

let component: any = null;
beforeEach(() => {
  component = document.createElement('div');
  document.body.append(component);
});

afterEach(() => {
  unmountComponentAtNode(component);
  component.remove();
  component = null;
});

describe('Content', () => {
  const sidebar = <div>sidebar</div>;
  const content = <div>content</div>;
  it('should render with sidebar', () => {
    act(() => {
      render(<Content sidebar={sidebar}>{content}</Content>, component);
    });

    expect(
      pretty(component.innerHTML)
    ).toMatchSnapshot();
  });

  it('should render without sidebar', () => {
    act(() => {
      render(<Content>{content}</Content>, component);
    });

    expect(
      pretty(component.innerHTML)
    ).toMatchSnapshot();
  });
});
