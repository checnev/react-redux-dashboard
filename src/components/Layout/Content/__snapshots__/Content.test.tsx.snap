// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render with sidebar 1`] = `
"<div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\">
  <div class=\\"MuiGrid-root MuiGrid-container\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3\\">
      <div>sidebar</div>
    </div>
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9\\">
      <div>content</div>
    </div>
  </div>
</div>"
`;

exports[`Content should render without sidebar 1`] = `
"<div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\">
  <div class=\\"MuiGrid-root MuiGrid-container\\">
    <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12\\">
      <div>content</div>
    </div>
  </div>
</div>"
`;
